xbmc/dialogs/GUIDialogSmartPlaylistRule.cpp

```diff
@@ -465,15 +426,83 @@ void CGUIDialogSmartPlaylistRule::UpdateButtons()
     labels.push_back(OperatorLabel(CDatabaseQueryRule::OPERATOR_DOES_NOT_EQUAL));
     break;
   }
+  return labels;
+}
 
-  SET_CONTROL_LABELS(CONTROL_OPERATOR, m_rule.m_operator, &labels);
+void CGUIDialogSmartPlaylistRule::OnCancel()
+{
+  m_cancelled = true;
+  Close();
+}
+
+void CGUIDialogSmartPlaylistRule::OnField()
+{
+
+  std::vector<Field> fields = CSmartPlaylistRule::GetFields(m_type);
+  CGUIDialogSelect* dialog = (CGUIDialogSelect*)g_windowManager.GetWindow(WINDOW_DIALOG_SELECT);
+  dialog->Reset();
+  dialog->SetHeading(CVariant{20427});
+  int selected = -1;
+  for (unsigned int i = 0; i < fields.size(); i++)
```

Comment:
i need `i`... and I would need sth like `for count, element in enumerate(fields)` How do I do that in Cpp? :)